blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorManager.java

```diff
@@ -64,33 +51,24 @@ public void requestCodeGeneration(List<String> blockDefinitionsJsonPaths,
                                       CodeGeneratorCallback codeGenerationCallback) {
 
         if (workspace.hasBlocks()) {
-            if (isBound()) {
-                try {
-                    final StringOutputStream serialized = new StringOutputStream();
-                    workspace.serializeToXml(serialized);
+            try {
+                final StringOutputStream serialized = new StringOutputStream();
+                workspace.serializeToXml(serialized);
+                CodeGenerationRequest codeGenerationRequest =
+                    new CodeGenerationRequest(
+                        serialized.toString(),
+                        codeGenerationCallback,
+                        blockDefinitionsJsonPaths,
+                        generatorsJsPaths);
+                mCodeGenerationConnection.requestCodeGeneration(codeGenerationRequest);
```

Comment:
Revisiting this code, the new CodeGenerationRequest() feels odd here, in that it doesn't seem to provide value to this class.

Option #1: It could be passed constructed previously, and then passed in.  This eliminates the need for workspace in this class, and could be returned as part of the callback.  I prefer this for that last point, but that would be a breaking change to the callback interface.

Option #2: The CodeGenerationRequest could be buried deeper, passing each of the parameters into the lazy service for construction there.